cli/install/install.go

```diff
@@ -0,0 +1,776 @@
+	if !checkProgrammDependencies("tt") && !flags.Force {
+		return nil
+	}
+
+	path, err := os.MkdirTemp("", "tt_install")
+	if err != nil {
+		return err
+	}
+	os.Chmod(path, 0777)
+
+	if !flags.Noclean {
+		defer os.RemoveAll(path)
+	}
+
+	// Download tt.
+	if flags.Local {
+		if checkExisting("tt", distfiles) {
+			log.Infof("Local files found, installing from them...")
+			localPath, _ := util.JoinAbspath(distfiles, "tt")
+			err = util.CopyDir(localPath, path)
+			if err != nil {
+				printLog(logFile.Name())
+				return err
+			}
+			util.ExecuteCommand("git", flags.Verbose, logFile, path, "checkout", "master")
+		} else {
+			return fmt.Errorf("Can't find distfiles directory.")
+		}
+	} else {
+		log.Infof("Downloading tt...")
+		err = downloadRepo(TTLink, "master", path, logFile, flags.Verbose)
+	}
+
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+	// Build tt.
+	log.Infof("Building tt...")
+	err = util.ExecuteCommand("mage", flags.Verbose, logFile, path,
+		"build")
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+
+	// Copy binary.
+	log.Infof("Copying executable...")
+	if _, err := os.Stat(dst); os.IsNotExist(err) {
+		err = os.Mkdir(dst, 0777)
+		if err != nil {
+			return fmt.Errorf("Unable to create bindir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		os.Remove(filepath.Join(dst, version))
+	}
+	err = util.CopyFilePreverse(filepath.Join(path, "tt"), filepath.Join(dst, version))
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+
+	// Set symlink.
+	err = util.CreateSymLink(version, dst, "tt", true)
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+	log.Infof("Done.")
+	if flags.Noclean {
+		log.Infof("Artifacts can be found at: " + path)
+	}
+	return nil
+}
+
+func checkExistingTarantool(version string, bindst string, headerDir string,
+	flags InstallationFlag) error {
+	var err error
+	if checkExisting(version, bindst) && !flags.Reinstall {
+		log.Infof("%s version of tarantool already exists, updating symlink...", version)
+		err := util.CreateSymLink(version, bindst, "tarantool", true)
+		log.Infof("Done")
+		return err
+	} else if checkExisting(version, bindst) && flags.Reinstall {
+		log.Infof("%s version of tarantool already exists, removing binary...", version)
+		os.RemoveAll(filepath.Join(bindst, version))
+	}
+
+	if checkExisting(version, headerDir) && !flags.Reinstall {
+		log.Infof("%s version of tarantool headers already exists, updating symlink...", version)
+		err := util.CreateSymLink(version, headerDir, "tarantool", true)
+		log.Infof("Done")
+		return err
+	} else if checkExisting(version, bindst) && flags.Reinstall {
+		log.Infof("%s version of tarantool binary already exists, removing headers...", version)
+		os.RemoveAll(filepath.Join(headerDir, version))
+	}
+	return err
+}
+
+// installTarantool installs selected version of tarantool.
+func installTarantool(version string, bindst string, headerDir string, flags InstallationFlag,
```

Comment:
Pls, don't use abbreviations like bindst. It is unclear.

Reply:
Changed